dice-game-ui-angular/src/app/services/local-game-storage-service.ts

```diff
@@ -1,33 +1,70 @@
 import { Injectable } from '@angular/core';
 import { Guid } from 'guid-typescript';
-import { GameEngine, SimpleGameEngine, Size } from 'engine';
-import { GameStorageService } from './game-storage-service';
+import { GameEngine, SimpleGameEngine } from 'engine';
+import { GameStorageService, SavedGameEngine } from './game-storage-service';
 
 @Injectable()
 export class LocalGameStorageService implements GameStorageService {
+  private readonly savedGamesKey = 'saved-games';
+
   createGame(gameEngine: GameEngine): Guid {
-    const guid = Guid.create();
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
-    return guid;
+    const savedGameEngine = new SavedGameEngine(gameEngine);
+    const curSavedGames = this.getAllSavedGames();
+    if (curSavedGames.filter((savedGame) => savedGame.guid === savedGameEngine.guid).length === 0) {
+      curSavedGames.push(savedGameEngine);
+      localStorage.setItem(this.savedGamesKey, JSON.stringify(curSavedGames));
+    }
+    return Guid.parse(savedGameEngine.guid);
   }
 
   saveGame(gameEngine: GameEngine, guid: Guid): Guid {
-    if (!localStorage.getItem(guid.toString())) {
-      throw new Error('The saved game was not found in the store');
+    const allSavedGames = this.getAllSavedGames();
+    if (!allSavedGames) {
```

Comment:
This condition is impossible, because the method will always return at least an empty array.

Reply:
Is that not working as a "length === 0" check?

Reply:
Definitely not. It checks for null, undefined, 0, empty string and NaN.

Reply:
Gor it. Changed